ros2doctor/ros2doctor/verb/hello.py

```diff
@@ -71,125 +77,158 @@ def add_arguments(self, parser, cli_name):
             '-1', '--once', action='store_true', default=False,
             help='Publish and multicast send for one emit period then exit; used in test case.')
 
-    def main(self, *, args):
-        global summary_table
-        summary_table = SummaryTable()
-        rclpy.init()
-        executor = SingleThreadedExecutor()
-        pub_node = Talker(args.topic, args.emit_period)
-        sub_node = Listener(args.topic)
-        executor.add_node(pub_node)
-        executor.add_node(sub_node)
-        try:
-            prev_time = time.time()
-            # pub/sub thread
-            exec_thread = threading.Thread(target=executor.spin)
-            exec_thread.start()
-            while True:
-                if (time.time() - prev_time > args.print_period):
-                    summary_table.format_print_summary(args.topic, args.print_period)
-                    summary_table.reset()
-                    prev_time = time.time()
-                # multicast threads
-                send_thread = threading.Thread(target=_send, kwargs={'ttl': args.ttl})
-                send_thread.daemon = True
-                receive_thread = threading.Thread(target=_receive)
-                receive_thread.daemon = True
-                receive_thread.start()
-                send_thread.start()
-                time.sleep(args.emit_period)
-                if args.once:
-                    return summary_table
-        except KeyboardInterrupt:
-            pass
-        finally:
-            executor.shutdown()
-            rclpy.shutdown()
-            pub_node.destroy_node()
-            sub_node.destroy_node()
-
-
-class Talker(Node):
-    """Initialize talker node."""
-
-    def __init__(self, topic, time_period, *, qos=10):
-        node_name = NODE_NAME_PREFIX + '_talker'
-        super().__init__(node_name)
-        self._i = 0
-        self._pub = self.create_publisher(String, topic, qos)
-        self._timer = self.create_timer(time_period, self.timer_callback)
-
-    def timer_callback(self):
+    def main(self, *, args, summary_table=None):
+        if summary_table is None:
+            summary_table = SummaryTable()
+        with DirectNode(args, node_name=NODE_NAME_PREFIX + '_node') as node:
+            publisher = HelloPublisher(node, args.topic, summary_table)
+            HelloSubscriber(node, args.topic, summary_table)
```

Comment:
who is keeping a reference to this object and keeping it alive?

PS: I know that currently nodes keep a reference of all created subscriptions. I wouldn't rely on that implementation detail, it might change in the future.

Reply:
That's fair. I dropped it to avoid `unused variable` linter errors. I guess I can use `#noqa` instead.